pkg/kubelet/kubelet.go

```diff
@@ -232,12 +232,13 @@ func (kl *Kubelet) GarbageCollectLoop() {
 }
 
 func (kl *Kubelet) getUnusedImages() ([]string, error) {
-	kl.pullLock.Lock()
-	defer kl.pullLock.Unlock()
 	return dockertools.GetUnusedImages(kl.dockerClient)
 }
 
 func (kl *Kubelet) GarbageCollectImages() error {
+	kl.pullLock.Lock()
```

Comment:
I'm a bit worried that this will cause blips for new pods trying to start, but I guess we can solve that later.  Maybe a TODO here along the lines of "minimize the impact on new pulls by releasing the lock and re-validating between each removal"